### ISSUE TITLE
LIST BUCKET response is not correct for rclone serve s3 over blomp drive

### ISSUE THREAD

**Issue by diyism** [rclone/rclone#7500]
$ rclone version
```
rclone v1.65.0
- os/version: debian kali-rolling (64 bit)
- os/kernel: 6.1.0-kali9-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.21.4
- go/linking: static
- go/tags: none
```

My ~/.config/rclone/rclone.conf:
```
[blomp]
type = swift
user = me@gmail.com
key = 123456
auth = https://authenticate.blomp.com
tenant = storage
auth_version = 2
no_chunk = true
```

First I use rclone serve s3 on local folder:
`$ rclone serve s3 -vv --addr=127.0.0.1:8100 ./storage1`

Run this curl as a s3 client command:
```
$ curl "http://127.0.0.1:8100/rclone_serve_s3?prefix=test.sqlite/generations/&delimiter=/"
<?xml version="1.0" encoding="UTF-8"?>
<ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Name>rclone_serve_s3</Name>
  <IsTruncated>false</IsTruncated>
  <Delimiter>/</Delimiter>
  <Prefix>test.sqlite/generations/</Prefix>
  <MaxKeys>1000</MaxKeys>
  <CommonPrefixes>
    <Prefix>test.sqlite/generations/8734e9321417f1f4</Prefix>
  </CommonPrefixes>
  <Marker></Marker>
</ListBucketResult>
```

It works very well.

Then I use blomp swift:
`$ rclone serve s3 -vv --addr 127.0.0.1:8100 blomp:me@gmail.com`

And I run the same curl as a s3 client command:
```
$ curl "http://127.0.0.1:8100/rclone_serve_s3?prefix=test.sqlite/generations/&delimiter=/"
<?xml version="1.0" encoding="UTF-8"?>
<ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Name>rclone_serve_s3</Name>
  <IsTruncated>false</IsTruncated>
  <Delimiter>/</Delimiter>
  <Prefix>test.sqlite/generations/</Prefix>
  <MaxKeys>1000</MaxKeys>
  <CommonPrefixes>
    <Prefix>test.sqlite/generations/8734e9321417f1f4/snapshots/00000002.snapshot.lz4</Prefix>
  </CommonPrefixes>
  <CommonPrefixes>
    <Prefix>test.sqlite/generations/8734e9321417f1f4/wal/00000001_00001038.wal.lz4</Prefix>
  </CommonPrefixes>
  <CommonPrefixes>
    <Prefix>test.sqlite/generations/8734e9321417f1f4/wal/00000002_00000000.wal.lz4</Prefix>
  </CommonPrefixes>
  <Marker></Marker>
</ListBucketResult>
```

It seems this response is not correct, it should be the same with the former "test.sqlite/generations/8734e9321417f1f4".

**Comment by ncw** [comment]
I think I fixed this already in this branch - can you give it a try?

https://beta.rclone.org/branch/fix-serve-s3/v1.66.0-beta.7564.ad491d1cf.fix-serve-s3/

**Comment by diyism** [comment]
Great,
I've tested the https://beta.rclone.org/branch/fix-serve-s3/v1.66.0-beta.7564.ad491d1cf.fix-serve-s3/rclone-v1.66.0-beta.7564.ad491d1cf.fix-serve-s3-linux-amd64.deb
I confirms that it has fixed this issue, thanks.
